src/compas/datastructures/assembly/assembly.py

```diff
@@ -78,7 +88,7 @@ def add_part(self, part, key=None, **kwargs):
         """
         key = self.graph.add_node(key=key, part=part, **kwargs)
         part.key = key
-        self.parts[part.guid] = part
+        self._parts[part.guid] = part
```

Comment:
i guess the point is to keep track of parts globally. so maybe we should check if a part already exists in the assembly before adding it?

Reply:
good point, added at https://github.com/compas-dev/compas/pull/946/commits/27729ed3a913feb1ce81fd1ec9418b27de302d79